app/jobs/csv_export_job.rb

```diff
@@ -0,0 +1,65 @@
+require 'csv'
+
+class CsvExportJob < ActiveJob::Base
+  queue_as :csv_jobs
+ 
+  def perform(csv_export)
+    ActiveRecord::Base.connection_pool.with_connection do
+      create_export_folder(csv_export)
+      export_task(csv_export)
+      cleanup_downloaded
+    end
+  end
+  
+  private
+  
+  def cleanup_downloaded
+    # TODO delete files with no record associations
+    CsvExport.old.each do |csv_export|
+      csv_export.destroy
+    end
+  end
+  
+  def export_task(csv_export)
+    csv_export.status! :started
+    deploy_csv_export(csv_export)
+    notify_of_creation(csv_export)
+  rescue IOError, ActiveRecord::ActiveRecordError => e
+    csv_export.delete_file
+    csv_export.status! :failed
+    Rails.logger.error("Export #{csv_export.id} failed with error #{e}")
+    Airbrake.notify(e, error_message: "Export #{csv_export.id} failed.")
+  end
+  
+  def notify_of_creation(csv_export)
+    CsvMailer.created_email(csv_export).deliver_now if csv_export.email.present?
+    csv_export.status! :finished
```

Comment:
these 2 lines don't have anything to do with 'notifying' ... maybe better in the method above